---
title: Microbial population shifts during disturbance induced foaming in anaerobic digestion of primary and activated sludge - Figure 7
output:
  html_document:
    df_print: paged
editor_options:
  chunk_output_type: console
---

## Load packages and data

```{r, message=FALSE, warning=FALSE}
options("lubridate.week.start" = 3)
library(tidyverse)
theme_set(theme_bw())  
library(ggpubr)
library(lubridate)
library(ggrepel)
library(phyloseq)
library(ggtreeExtra)
library(ggtree)
library(ggnewscale)
`%notin%` = Negate(`%in%`)
# Prevalence table function
# get overview of abundances, mean prevalence is the mean 'appearance' of ASVs of the taxon across all samples
prevalencedf <- dget("./custom-functions/myprevalencetablefunction.R")
mytreefunction <- dget("./custom-functions/mytreefunction_simple.R")
startoperation <- ymd("2023-03-01")
cols <- c("#4D98AC", "#985C64", "#ff0000", "purple", "#C4A484")
names(cols) <- c("Control", "Treatment", "Full-scale", "PSTWAS", "Foam")

## LOAD MASTERDATA
# change this path to where you store the .csv file. you can provide an absolute path like this. Check the path syntax for Windows as it might be different. 
masterdata <- read.csv("./data/masterdataProject1C.csv")
masterdata <-  masterdata %>%
  mutate(Date = lubridate::dmy(as.character(Date)) )
# Make AD, Treatment and SludgeType = factors
masterdata$AD <- factor(masterdata$AD, levels = c("AD1", "AD2","AD3", "AD4","AD5", "AD6", "Full-scale", "PSTWAS"))
masterdata$SludgeType <- factor(masterdata$SludgeType, levels = c("Control", "Treatment", "Foam", "Full-scale", "PSTWAS"))
masterdata$Treatment <- factor(masterdata$Treatment, levels = c("Control", "Treatment", "Full-scale", "PSTWAS"))
masterdata$Period <- factor(masterdata$Period, levels = c("Converging", "SteadyState", "Glycerol", "Inhibition", "Recovery/Feedingpause", "Recovery/Foaming", "Recovery/Postfoam", "SteadyState/Postfoam"))
masterdata <- masterdata %>% 
mutate(across(Observed:pielou_ev, ~as.numeric(.)))


## TREE DATA
# BIN ABUNDANCES FOR PHYLOSEQ 
bindata <- data.frame(read_tsv("./data/18.1CRMITAD.bintableforR.tsv"))
tpm <- bindata %>% dplyr::select(`Bin.ID`, Completeness, Contamination, starts_with("TPM")) %>% column_to_rownames("Bin.ID")
completnessdf <- tpm %>% dplyr::select(Completeness, Contamination)
completnessdf <- completnessdf %>% 
  mutate(Quality = case_when((Completeness > 90 & Contamination < 5) ~ "High",
                             (Completeness > 50 & Contamination < 10) ~ "Medium",
                             .default = "Low"))
completnessdf$Quality <- factor(completnessdf$Quality, levels = c("Low","Medium","High"))
tpm <- tpm %>%  dplyr::rename(S561toS566_AD.AD6 = TPM.Chris_11.S561toS566_AD.AD6,
                           S1001_3_4_AD1_3_4 = TPM.Chris_12.S1001_3_4_AD1_3_4,
                           S1002_5_6_AD2_5_6 = TPM.Chris_13.S1002_5_6_AD2_5_6,
                           S1147_49_50_AD1_3_4 = TPM.Chris_14.S1147_49_50_AD1_3_4,
                           S1148_51_52_AD2_5_6 = TPM.Chris_15.S1148_51_52_AD2_5_6,
                           S1154_57_58_AD2_5_6 = TPM.Chris_16.S1154_57_58_AD2_5_6) %>% 
                dplyr::select(-Completeness)

taxa <-  bindata %>% dplyr::select(`Bin.ID`, TaxGTDB)  %>% 
  separate(TaxGTDB, sep=";", c("Kingdom","Phylum","Class","Order","Family","Genus","Species")) %>% column_to_rownames("Bin.ID")
taxa$Kingdom <-  str_replace_all(taxa$Kingdom, "d__", "")
taxa$Phylum <-  str_replace_all(taxa$Phylum, "p__", "") 
taxa$Class <-  str_replace_all(taxa$Class, "c__", "") 
taxa$Order   <-  str_replace_all(taxa$Order, "o__", "") 
taxa$Family <-  str_replace_all(taxa$Family, "f__", "") 
taxa$Genus <-  str_replace_all(taxa$Genus, "g__", "") 
taxa$Species <-  str_replace_all(taxa$Species, "s__", "") 
taxa[taxa == ""] <- NA

## TREE FILE FOR PHYLOSEQ
tree <- ape::read.tree("./data/1CRMITAD.nw")
old <- data.frame(tree$tip.label) %>% arrange(desc(tree.tip.label))
old$tip.label2 <- substr(old$tree.tip.label, start = 1, stop = 12)
new <- data.frame(rownames(taxa)) %>% arrange(desc(rownames.taxa.))
new$tip.label2 <- substr(new$rownames.taxa., start = 1, stop = 12)
tips <- old %>% left_join(new, by = "tip.label2") %>% dplyr::rename(old = "tree.tip.label", new = "rownames.taxa.")
tree$tip.label[match(tips$old, tree$tip.label)] <- tips$new

tiplabeldf <- taxa %>%  mutate(Label = coalesce(Species, Genus, Family, Order, Class, Phylum, Kingdom)) %>% rownames_to_column("ID") %>% 
  filter(ID %in% c(tips$new)) 
#tiplabeldf <- tiplabeldf %>% dplyr::select(ID, Phylum, Label) %>% tidyr::unite(col = "Label", sep=";", Phylum:Label)
tiplabeldf <- tiplabeldf %>% dplyr::select(ID, Label) 
#taxa <- taxa %>% filter(Phylum != "")  # remove one bin without classification (Bacteria)
rm(new, old, tips)


## METADATA SUBSET FOR PHYLOSEQ
# merge individual samples into the pooled samples - mean values
df <- masterdata %>% 
   as_tibble() %>% dplyr::filter(SampleID.DNA %in% c("S1001_AD1", "S1003_AD3", "S1004_AD4",
                                                     "S1002_AD2","S1005_AD5","S1006_AD6",
                                                     "S1147_AD1", "S1149_AD3", "S1150_AD4",
                                                     "S1148_AD2", "S1151_AD5", "S1152_AD6",
                                                     "S1154_AD2","S1157_AD5","S1158_AD6"
                                                     )) %>% column_to_rownames("SampleID.DNA") 
df <- data.frame(df %>% group_by(Date, SludgeType ) %>% summarise_each(funs(mean)))
rownames(df) <- c("S1001_3_4_AD1_3_4", "S1002_5_6_AD2_5_6","S1147_49_50_AD1_3_4", "S1148_51_52_AD2_5_6", "S1154_57_58_AD2_5_6")
df$Period <- c("Inhibition","Inhibition","Recovery/Foaming","Recovery/Foaming", "Recovery/Foaming")
df$AD <- c("AD1.3.4","AD2.5.6","AD1.3.4","AD2.5.6", "AD2.5.6")

# there was also 1 samples pooled for all 6 digesters 
df1 <- masterdata %>% 
   as_tibble() %>% dplyr::filter(SampleID.DNA %in% c("S561_AD1", "S562_AD2", "S563_AD3",
                                                     "S564_AD4","S565_AD5","S566_AD6"
                                                     )) %>% column_to_rownames("SampleID.DNA") 
df1 <- data.frame(df1 %>% group_by(Date ) %>% summarise_each(funs(mean)))
rownames(df1) <- c("S561toS566_AD.AD6")
df1$Period <- c("SteadyState")
df1$AD <- c("AD1.2.3.4.5.6")

# not pooled samples 
df2 <- masterdata %>% 
   as_tibble() %>% dplyr::filter(SampleID.DNA %in% c("F5","F55","F110","F124","F179", 
                                                     "S119_full","S370_full", "S1273_ADFull", "S1554_ADFull", "S1813_ADFull"
                                                     )) %>% 
   column_to_rownames("SampleID.DNA")

#combined the dfs
metadata <- rbind(df,df1, df2)
rownames(metadata)
metadata <- metadata %>% rownames_to_column("ID") %>% 
  mutate("SampleID" = ID) %>% column_to_rownames("ID")  %>% 
  mutate("SampleID" = factor(SampleID, levels = c("S561toS566_AD.AD6", "S1001_3_4_AD1_3_4", "S1002_5_6_AD2_5_6", "S1147_49_50_AD1_3_4", "S1148_51_52_AD2_5_6", "S1154_57_58_AD2_5_6",   "F5",    "F55", "F110","F124", "F179", "S119_full", "S370_full","S1273_ADFull", "S1554_ADFull","S1813_ADFull")))

metadata$ramaciottieID <- str_replace_all(metadata$SampleID, c( "F5"="Chris_01", "F55"="Chris_02", "F110"="Chris_03","F124" = "Chris_04",
                                      "F179" = "Chris_05", "S119.full" = "Chris_06", "S370.full" = "Chris_07", "S1273_ADFull" = "Chris_08",
                                      "S1554_ADFull" = "Chris_09", "S1813_ADFull" = "Chris_10", "S561toS566_AD.AD6" = "Chris_11", "S1001_3_4_AD1_3_4" = "Chris_12",
                                      "S1002_5_6_AD2_5_6" = "Chris_13", "S1147_49_50_AD1_3_4" = "Chris_14", "S1148_51_52_AD2_5_6" = "Chris_15", "S1154_57_58_AD2_5_6" = "Chris_16"))

metadata <- metadata %>% mutate(VFAs = rowSums(select(., contains("acid"))) )
metadata[c("S561toS566_AD.AD6"),c("SludgeType")] <- "Control"
rm(df, df1, df2)

ps <- phyloseq(
  otu_table(tpm, taxa_are_rows = T), 
  sample_data(metadata),
  tax_table(as.matrix(taxa)),
  phy_tree(tree)
)

tpm <- tpm %>% rename(SteadyState = "S561toS566_AD.AD6",
                 Control15May24 = "S1001_3_4_AD1_3_4",
                Treatment15May24 = "S1002_5_6_AD2_5_6",
              Control02June24 = "S1147_49_50_AD1_3_4",
              Treatment02June24 = "S1148_51_52_AD2_5_6",
              Foam02June24 = "S1154_57_58_AD2_5_6") %>% 
  dplyr::select(SteadyState, Control15May24,  Control02June24, Treatment15May24, Treatment02June24, Foam02June24)

## COLOURS
colstree <- readRDS("./data/treecolsvector") # from 16S data phyla
treecols2 <- colstree
df <- psmelt(ps) %>% 
  dplyr::filter(Abundance > 0) %>% 
  arrange(desc(Abundance))
names <- df$Phylum  %>% unique() # metagenome names (from GenBank nr)
#phyla <- prevalencedf(ps, Phylum)
treecols2 <- colstree
bacillota <- c("#3BA0FDFF", "#3BA0FDFF", "#3BA0FDFF", "#3BA0FDFF", "#3BA0FDFF","#3BA0FDFF", "#3BA0FDFF")
names(bacillota) <- c("Bacillota_A","Bacillota_B","Bacillota_C","Bacillota_D", "Bacillota_E", "Bacillota_G","Bacillota_I")
desulfob <- c("#1AE4B6FF") 
names(desulfob) <- c("Desulfobacterota_G") 
names(treecols2) <- c("Pseudomonadota","Bacillota" , "Bacteroidota", "Actinomycetota", "",
                      "Planctomycetota", "Verrucomicrobiota","Acidobacteriota", 
                      "Desulfobacterota","Patescibacteria","Spirochaetota", "Synergistota", "Halobacteriota",
                     "Bdellovibrionota", "Campilobacterota", "Cloacimonadota",
                     "Cyanobacteriota", "Dependentiae", "Euryarchaeota", "Fibrobacterota",
                     "Thermotogota", "Myxococcota","", "Caldatribacteriota", 
                     "Caldisericota","Crenarchaeota",
                     "Elusimicrobiota","Fusobacteriota","Gemmatimonadota", "Hydrogenedentota", 
                     "Latescibacterota", "Nitrospirota","Sumerlaeota", "CK-2C2-2", "Coprothermobacterota", "Dictyoglomota", 
                     "SAR324_clade(Marine_group_B)", "Sva0485", "Thermoplasmatota", "", "", "Methanobacteriota", "WS4", "Atribacterota",
                     "Armatimonadota","Chloroflexota")
treecols <- c(treecols2, bacillota, desulfob)
```


## Figure 7 - tree bins

```{r fig.height=10, fig.width=10, message=FALSE, warning=FALSE}
filtervec <- completnessdf %>% dplyr::filter(Quality %in% c("Medium", "High"))

ps.phylo <- ps
# Filter the phyloseq object
ps.phylo  <- prune_taxa(taxa_names(ps.phylo ) %in% rownames(filtervec), ps.phylo )

taxa_names(ps.phylo)
# use custom prevalence table function and select phyloseq object and phylum-level
taxa.df.phylum <- prevalencedf(ps.phylo, Phylum)
# set seed differently will change the random sampling of the colours
# set.seed(1)
# randomly sample colours from this colour scheme
# Use the same tree cols for all figures 
ordervec <- (taxa.df.phylum %>% 
            dplyr::filter(Total_abundance >= 22300.611) %>% 
              arrange(Phylum))$Phylum
# cat(paste(shQuote(ordervec, type="cmd"), collapse=", ")) # create comma separated vector

# If you prefer number of ASVs per each Phylum to be shown in legend:
# create a label vector to change the names in the legend. Needs to be in correct order. 
labelvec <- (taxa.df.phylum %>% 
            dplyr::filter(Total_abundance >= 22300.611) %>% 
              arrange(Phylum) %>% 
            mutate(Phylum = paste0(Phylum, " (", n, ")")))$Phylum

p <- mytreefunction(ps.phylo, 
         layout = "fan",
         branch.length = "none",
         offset.text = 5,
         pwidth = 0.05,
         open.angle = 25) + 
         scale_colour_manual("Phylum (number of MAGs)",
                              values = treecols, 
                              limits = ordervec,
                              label = labelvec) +
      guides(colour = guide_legend(override.aes = list(size = 5), 
                               keywidth = 0.5, 
                                keyheight = 0.5,
                               ncol = 1,
                               order = 1)) +
  theme(
    legend.text = element_text(size = 12) # Set legend text size
  )
      ggtitle("Bins RMIT AD 01 May, 15 May, 02 June") 
p <- rotate_tree(p, 100)

### Heatmap with completeness 
p <- p %<+% tiplabeldf + geom_tiplab2(size = 3.75, offset = 6.5, aes(label = Label)) +
  xlim(0, 36)

p <- gheatmap(p, completnessdf["Quality"], offset=0.01, width=.03, font.size = 2.5, colnames = TRUE, 
          colnames_angle=95, colnames_offset_y = .25, custom_column_labels = ""
         ) +
    scale_fill_manual(name="MAG quality", values = c("#82E0AA", "#1D8348"),
                      limits = c("Medium", "High"),
                      label = c("Medium", "High"),
                      guide = guide_legend(order = 2)) 

#Options
# "Low (<50% complete or >10% contaminated)", "Medium (>50% complete & <10% contaminated)", "High (>90% complete & <5% contaminated)"
# ggsave(plot = ph, "./Figures/bintree.png", height=35, width=35, units='cm', dpi=600, limitsize = FALSE)

### Heatmap with TPM
p2 <- p + new_scale_fill()
p2 <- gheatmap(p2, as.matrix(tpm), offset=0.5, width=0.25, font.size = 3, hjust = 1,
         colnames_angle=-10, colnames_offset_y = 0, colnames_position = "top", 
         custom_column_labels = c("Steady State - Day 61", "Control - Day 75", "Control - Day 93", 
                                  "Treatment - Day 75", "Treatment - Day 93", "Foam - Day 93")
         ) +
  
     scale_fill_viridis_c("TPM", option = "rocket", direction = -1, begin = 0, end = 1)  
p2

ggsave(plot = p2, "./Figures/bintree_tpm_daysop.png", height=30, width=35, units='cm', dpi=300, limitsize = FALSE)
ggsave(plot = p2, "./Figures/bintree_tpm_daysop.tiff", height=30, width=35, units='cm', dpi=300, limitsize = FALSE)

```
